ui/src/app/job-list/table/table.component.ts

```diff
@@ -200,4 +200,19 @@ export class JobsTableComponent implements OnInit {
       this.jobs.forEach(j => this.selection.select(j));
     }
   }
+
+  updateCheckBoxSelection(clickedJob: QueryJobsResult, event: MouseEvent): void {
+    // if the user has shift-clicked on a job, find the last selected job and
+    // select all jobs between the two
+    if (event.shiftKey && this.selection.selected) {
+      const lastJobClicked = this.selection.selected[this.selection.selected.length - 1];
+      const [from, to] = [
+        this.jobs.findIndex(j => j.id === lastJobClicked.id),
+        this.jobs.findIndex(j => j.id === clickedJob.id)
+      ].sort();
+      for (let i = from; i <= to; i++) {
```

Comment:
I thought that you didn't want to toggle the clicked checkbox? Would this double-toggle the checkbox you clicked?

Reply:
It's not toggling, it's marking them as selected; if they are already selected, there will be no change.

Reply:
But isn't `change` still running on the target checkbox? Is this depending on the `change` callback happening before `click`?

Reply:
when I tested it, `change` was not triggered when handled by the `click` handler

Reply:
Interesting.. that's not what I observe when I test out the mat-checkbox in stackblitz. Are you sure it's not just calling `change`, then calling `click` (overriding the toggle effect)? Can you add logging locally? I'm not sure I understand why normal checkbox interactions would still be working if change is not being triggered.

Anyways - unit tests may reveal more.